### PR TITLE
Code fix

### PR DESCRIPTION
**Pull request by sobuj53**
Code has been fixed by xerta555 Rom. Moved crond schedule to post boot entry.

### CODE REVIEW

**Comment by sobuj53** [comment]
please don't merge yet. I think it has some issue.

File: system/bin/Detach [hunk 1]
```diff
@@ -221,7 +221,7 @@ rm -f "$MAGMOD/apps_lists.txt"
 
 # Adding app
 add() {
-SERVICESH=$MAGMOD/service.sh
+#SERVICESH=$MAGMOD/service.sh
```
Comment (xerta555) [inline system/bin/Detach]: The script can't run propeply as the var for the `services.sh` file is declarated for each functions to be sure that everyting will working in any situation.

File: system/bin/Detach [hunk 2]
```diff
@@ -435,7 +435,7 @@ esac
 
 # Removing app
 rem() {
-SERVICESH=$MAGMOD/service.sh
+#SERVICESH=$MAGMOD/service.sh
```
Comment (xerta555) [inline system/bin/Detach]: Same comment

File: compatibility.txt [hunk 3]
```diff
@@ -3,6 +3,9 @@
 ps_accounts=$("$MODDIR/sqlite" $PS_DATA_PATH "SELECT account FROM ownership" | sort -u | wc -l)
 ps_accounts_final=$((ps_accounts+1))
 
+MAGMOD=/data/adb/modules/Detach
```
Comment (xerta555) [inline compatibility.txt]: I would prefere to have common vars at the beginning of the `compatibility.txt` file, to avoid any scripting issue(s)

**Comment by sobuj53** [comment]
Can you please take a look at [sobuj53-test-1](https://github.com/sobuj53/Detach/tree/sobuj53-test-1)? 
What is this test repo about?
- Well after you fixed the `compatibility.txt` file the `service.sh` was running in infinite loop because of `sh "$MAGMOD/service.sh" > "$MAGMOD/first_detach_result.txt" 2>&1` line in the service file. As it was misspelled like `services.sh` that's why we didn't observe this behavior earlier.
- I tried to fixed this behavior by creating a new `sh` file and calling it from `service.sh` only once.

**Comment by xerta555** [comment]
@hoypolli I have check your main.sh file and I'm not necessarily against adding a new sh file in the module, but I've never had a negative feedback when the script doesn't run when the device starts 🤷🏼‍♂️

File: Detach.txt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ 
+#Contacts 
+#Gmail 
+#Google App 
+#Google Plus 
+#Hangouts 
+#Phone 
+#Photos 
+YouTube
+#YouTube Music
```
Comment (xerta555) [inline Detach.txt:13]: It's useless to adding another app here if you don't edit the `customize.sh` file..

Comment (sobuj53) [reply Detach.txt:13]: I did not updated here

Comment (sobuj53) [reply Detach.txt:13]: in test repo. I'll push the change to main repo after completing the testing

Comment (xerta555) [reply Detach.txt:13]: > in test repo. I'll push the change to main repo after completing the testing

Np. Take your time.

**Comment by sobuj53** [comment]
I think the pull request is ready to merge. Please test and merge if you think it's okay

**Comment by sobuj53** [comment]
@xerta555 can you please test [Detach-test](https://github.com/sobuj53/Detach/tree/sobuj53-test) repo? I've updated sqlite3 to its latest version 3.35.5. Also added magisk sepolicy rules to work under fully selinux enforcing system. We can merge those changes if you find them ok.

**Comment by sobuj53** [comment]
@xerta555 hope this will fix any surprise error. Hopefully those are my final changes

**Comment by xerta555** [comment]
> @xerta555 hope this will fix any surprise error. Hopefully those are my final changes

I also hope, i have in mind to test internally on Telegram the last BETA test version, then from 10 positive feedbacks, migrate all the changes of the module to make the brand new version official.

File: customize.sh [hunk 5]
```diff
@@ -107,7 +107,7 @@ if [ ! -e "$CONF" ]; then
 
 	echo -e "!!! Your Detach.txt file does not exist !!!"
 	echo -e "!!! Copying Detach file to your storage root!!!"
-	cp -af "$TMPDIR/Detach.txt" "$DetachFile"
+	cp -fr "$TMPDIR/Detach.txt" "$DetachFile"
```
Comment (xerta555) [inline customize.sh]: Why using `-r` argument?

File: customize.sh [hunk 6]
```diff
@@ -126,7 +126,7 @@ grep -q '\.' "$TMPDIR/SYN_CONF.txt"; if [ $? -eq 0 ]; then
 	ui_print ""
 	echo -e "!- Your "$CONF" file contain error(s)"
 	ui_print "Now copying new Detach file, check and try again."
-	cp -afr "$TMPDIR/Detach.txt" "$DetachFile"
+	cp -fr "$TMPDIR/Detach.txt" "$DetachFile"
```
Comment (xerta555) [inline customize.sh:133]: Same here

File: customize.sh (hunk 5 above)
Comment (sobuj53) [reply customize.sh]: not really needed but it will make sure to overwrite the file if already there

File: customize.sh (hunk 6 above)
Comment (sobuj53) [reply customize.sh:133]: It's necessary in this case. As it's now necessary to overwrite already existing file.

**Pull request merged**